docs/architecture/blazor-for-web-forms-developers/data.md

```diff
@@ -1,22 +1,153 @@
+
+Once applied, you have a simple `Product` table in your database and some new classes added to the project that help manage the database schema.  You can find these generated classes, by default, in a new folder called `Migrations`.  When you make changes to the `Product` class or add more related classes you would like interacting with your database, you need to run the command-line commands again with a new name of the migration.  This command will generate another set of migration classes to update your database schema.
+
+### Data First
+
+For existing databases, you can generate the classes for Entity Framework Core.  The .NET command-line tools to scaffold the classes are similar to these two commands at the command line:
+
+```dotnetcli
+dotnet ef dbcontext scaffold "CONNECTION STRING" Microsoft.EntityFrameworkCore.SqlServer -c MyDbContext -t Product -t Customer
+```
+
+This command will connect to the database using the specified connection string and the `Microsoft.EntityFrameworkCore.SqlServer` provider.  Once connected, it will create a class called `MyDbContext` with supporting classes for the `Product` and `Customer` tables that were specified with the `-t` switches. There are many configuration options for this command to generate the class hierarchy appropriate for your database. You can find a complete reference in the [documentation for this command](https://docs.microsoft.com/ef/core/miscellaneous/cli/dotnet#dotnet-ef-dbcontext-scaffold).
+
+More information about [Entity Framework Core](https://docs.microsoft.com/ef/core/) can be found on the Microsoft Docs site.
+
+## Interacting with web services
+
+When ASP.NET was first released, SOAP services were the preferred way for web servers and clients to exchange data.  Much has changed since that time, and the preferred interactions with services have shifted to direct HTTP client interactions.  With ASP.NET Core and Blazor, you can register the configuration of your `HttpClient` in the Startup class's `ConfigureServices` method use that configuration wherever you need to interact with the HTTP endpoint.  Consider the following configuration code:
+
+```csharp
+
+services.AddHttpClient("github", c =>
+{
+    c.BaseAddress = new Uri("http://api.github.com/");
+    // Github API versioning
+    c.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json");
+    // Github requires a user-agent
+    c.DefaultRequestHeaders.Add("User-Agent", "BlazorWebForms-Sample");
+});
+
+```
+
+Whenever we need to access data from GitHub, we can create a client with a name of `github` and it will be configured with the appropriate base-address and headers set appropriately.  Inject the `IHttpClientFactory` into your Blazor components with the `@inject` directive or an `Inject` attribute on a property.  Create your named client and interact with services using this syntax:
+
+```cshtml
```

Comment:
```suggestion
```razor
```

Should this be `razor` instead?

Reply:
I tried selecting razor in VS Code autocomplete and it flipped to cshtml. Happy to change if that is the correct value